cypress/S3Monitor.py

```diff
@@ -0,0 +1,127 @@
+import boto3
+import json
+import os
+import quopri
+import time
+import re
+
+mailbox_bucket = 'opg-lpa-casper-mailbox' # this might get renamed once casper tests are turned off
+activation_emails_path = 'cypress/activation_emails'
+
+def assume_role_and_get_client():
+    sts = boto3.client(
+        'sts',
+        region_name='eu-west-1',
+    )
+
+    if os.getenv('CI'):
+        print("Assuming CI role")
+        role_arn = 'arn:aws:iam::050256574573:role/opg-lpa-ci'
+    else:
+        print("Assuming operator role")
+        role_arn = 'arn:aws:iam::050256574573:role/operator'
+
+    result = sts.assume_role(
+        RoleArn=role_arn,
+        RoleSessionName='session1',
+    )
+
+    s3Client = boto3.client(
+        's3',
+        aws_access_key_id=result['Credentials']['AccessKeyId'],
+        aws_secret_access_key=result['Credentials']['SecretAccessKey'],
+        aws_session_token=result['Credentials']['SessionToken']
+    )
+
+    return s3Client
+
+# Extract the plus part from emails of the form:
+# basename+pluspart@example.com
+def getPlusPartFromEmailAddress(email):
+    match = re.search("[^\+]\+(.+)@", email)
+    if match is None:
+        return ""
+    return match.group(1)
+
+def parseBody(bodyContent, subject, thetype, linkRegex):
+    regex = 'https:\/\/\S+' + linkRegex + '\/[a-zA-Z0-9]+'
+
+    match = re.search(regex, bodyContent)
+
+    if match is not None: 
+        s = match.start()
+        e = match.end()
+        activationLink = bodyContent[s:e]
+        print(f'{ thetype } link { activationLink }')
+
+        emailRegex = 'To: (.+)'
+
+        emailMatch = re.search(emailRegex, bodyContent)
+        if emailMatch is not None:
+            es = emailMatch.start()
+            ee = emailMatch.end()
+            toEmail = bodyContent[es:ee]
```

Comment:
If you just want the email address, you could do `toEmail = emailMatch.group(1)` here, I think (no need for ee and es).

Reply:
Sure, that's more elegant :) just needed the email address, so , now done the emailMatch.group(1).  Looks to work fine.